content/components/dialog.mdx

```diff
@@ -0,0 +1,265 @@
+  </div>
+</Box>
+
+### Body region
+
+The body region is the main content of the Dialog. It can contain any content that is relevant to the task at hand. The body region is scrollable if the content exceeds the height of the Dialog.
+
+<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={5}>
+  <div>
+    <img width="650" alt="" src="https://user-images.githubusercontent.com/18661030/185506131-1436b06b-a8fb-4de8-b9b2-4715ab0a63d2.png"/>
+    <Caption>Body content will scroll vertically if the content exceeds the overall Dialog height.</Caption>
+  </div>
+  <div>
+    <img width="650" alt="" src="https://user-images.githubusercontent.com/18661030/185478320-2a4a3053-dca3-4048-b7c4-d5a39a83acd6.png" />
+    <Caption>The body may contain additional content such as banners and form controls.</Caption>
+  </div>
+</Box>
+
+### Footer region
+
+The footer region may be used to show confirmation actions, navigation links, or specialized actions.
+
+If the content area has overflow scrolling, a divider will be shown between the footer and content area. Otherwise, showing a divider is optional.
+
+<img src="https://user-images.githubusercontent.com/293280/175850630-9f3df284-c69b-4d16-9e57-ea04a0202d04.png" width="960" alt="" />
+<Caption>
+  Use buttons in a footer to let the user complete a task.
+</Caption>
+
+<Box display="grid" gridTemplateColumns={['1fr', null, null, null, '1fr 1fr']} gridGap={5}>
+  <div>
+    <img src="https://user-images.githubusercontent.com/293280/175850639-a77d537e-2807-47d2-b456-a298d5f44918.png" alt="" />
+    <Caption>Footer with small buttons.</Caption>
+  </div>
+  <div>
+    <img src="https://user-images.githubusercontent.com/293280/175850637-3b193552-b35f-4b07-b2d6-4ca546c8131a.png" alt="" />
+    <Caption>Footer with auxiliary create button.</Caption>
+  </div>
+</Box>
+
+### Backdrop
+
+Dialog displays with a **backdrop**, which dims the rest of the page. The backdrop is visible when the Dialog is centered on the page, or attached to a sides as a sheet.
+
+By default, clicking on the **backdrop** will dismiss the Dialog. However, if the Dialog contains a form that can have unsaved changes, the **backdrop** won't dismiss the Dialog, regardless of the state of the form.
+
+### Dimensions
+
+Dialogs should never extend beyond the viewport sizes, and may resize to fit smaller devices. The body contents should fit narrow viewports with a minimum width of `320px`. If the contents extend vertically beyond the specified height, the **body** region will scroll.
+
+The `height` and `width` may be set to `auto` which will set a minimum width of `192px` and respond to the contents of the Dialog. Additionally, an explicit `height` and `width` may be set with a specific set of options.
+
+<img width="1292" alt="" src="https://user-images.githubusercontent.com/18661030/185178528-e1c176ef-f407-40ab-8729-965e21d64f6d.png"/>
```

Comment:
I'm curious how folks feel about this image. We offer a large amount of size combinations, and finding a way to show that in a static image is tricky! More discussion [here](https://github.com/github/primer/pull/1202/files#r944065171). Open to suggestions!

Reply:
Here are some alternative options:

<img width="1292" alt="Size options for Dialog" src="https://user-images.githubusercontent.com/18661030/185514743-c27965cb-08f6-4236-8cec-663788e1f2dc.png">

This ones from @vdepizzol!

<img width="1216" alt="Size options for Dialog" src="https://user-images.githubusercontent.com/18661030/185514800-c2d231aa-11fb-4f51-a23c-4c54649ce0b3.png">

Reply:
This is great ✨ some thoughts/comments: 

- 👍  to add a width/height label to the X/Y axis
- 👍  to use a single purple color rather than the sequential palette. It's more clear when trying to understand what is a ✅ and a 🔴 combination.
- What about a legend along with the image with all the possible size combinations?
- Should we explain why we provide these combinations and not others? e.g, line length good practices

Reply:
This is great ✨ some thoughts/comments:

- 👍 to add a width/height label to the X/Y axis
- 👍 to use a single purple color rather than the sequential palette. It's clearer when looking for what is a ✅ and a 🔴 combination.
- What about a legend along with the image with all the possible size combinations?
- Should we explain why we provide these combinations and not others? e.g, line length good practices

Reply:
I think this has been solved with the newer sizing options? Marking as resolved.